letsencrypt/client/display/util.py

```diff
@@ -0,0 +1,410 @@
+
+        :param str message: title of menu
+        :param choices: Menu lines
+        :type choices: list of tuples (tag, item) or
+            list of descriptions (tags will be enumerated)
+
+        """
+        # Can take either tuples or single items in choices list
+        if choices and isinstance(choices[0], tuple):
+            choices = ["%s - %s" % (c[0], c[1]) for c in choices]
+
+        # Write out the message to the user
+        self.outfile.write(
+            "{new}{msg}{new}".format(new=os.linesep, msg=message))
+        side_frame = ("-" * 79) + os.linesep
+        self.outfile.write(side_frame)
+
+        # Write out the menu choices
+        for i, desc in enumerate(choices, 1):
+            self.outfile.write(
+                textwrap.fill("{num}: {desc}".format(num=i, desc=desc), 80))
+
+            # Keep this outside of the textwrap
+            self.outfile.write(os.linesep)
+
+        self.outfile.write(side_frame)
+
+    def _wrap_lines(self, msg):  # pylint: disable=no-self-use
+        """Format lines nicely to 80 chars
+
+        :param str msg: Original message
+
+        :returns: Formatted message respecting newlines in message
+        :rtype: str
+
+        """
+        lines = msg.splitlines()
+        fixed_l = []
+        for line in lines:
+            fixed_l.append(textwrap.fill(line, 80))
+
+        return os.linesep.join(fixed_l)
+
+    def _get_valid_int_ans(self, max_):
+        """Get a numerical selection.
+
+        :param int max: The maximum entry (len of choices), must be positive
+
+        :returns: tuple of the form (`code`, `selection`) where
+            `code` - str display exit code ('ok' or cancel')
+            `selection` - int user's selection
+        :rtype: tuple
+
+        """
+        selection = -1
+        if max_ > 1:
+            input_msg = ("Select the appropriate number "
+                         "[1-{max_}] then [enter] (press 'c' to "
+                         "cancel): ".format(max_=max_))
+        else:
+            input_msg = ("Press 1 [enter] to confirm the selection "
+                         "(press 'c' to cancel): ")
+        while selection < 1:
+            ans = raw_input(input_msg)
+            if ans.startswith("c") or ans.startswith("C"):
+                return CANCEL, -1
+            try:
+                selection = int(ans)
+                if selection < 1 or selection > max_:
+                    selection = -1
+                    raise ValueError
+
+            except ValueError:
+                self.outfile.write(
+                    "{0}** Invalid input **{0}".format(os.linesep))
+
+        return OK, selection
+
+
+def separate_list_input(input_):
+    """Separate a comma or space separated list.
+
+    :param str input_: input from the user
+
+    :returns: strings
+    :rtype: list
+
+    """
+    no_commas = input_.replace(",", " ")
+    # Each string is naturally unicode, this causes problems with M2Crypto SANs
```

Comment:
no_commas split takes care of all the whitespace before and after the list.  https://github.com/letsencrypt/lets-encrypt-preview/blob/revoker/letsencrypt/client/tests/display/util_test.py#L303